### PR TITLE
First draft of debugging docs.

### PR DESCRIPTION
**Pull request by isaacabraham**

### CODE REVIEW

File: docs/feature-debugging.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
+A powerful features of the SAFE Stack is that it supports combined client / server debugging in F#. This means that you can use a single IDE (in this case, [Visual Studio Code](https://code.visualstudio.com/)) to debug *both client and server code* in the **same IDE session**.
+
+As of v0.38 of the SAFE Template, new SAFE applications come pre-configured with build, launch and debugging support in VS Code.
+
+## Prerequisites
+1. **Install [Google Chrome](https://www.google.com/chrome/)**: Enables for client-side debugging.
```
Comment (theimowski) [inline docs/feature-debugging.md]: Remove extra `for`

File: docs/feature-debugging.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+1. **Install the [Redux Dev Tools](https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd?hl=en)** extension for Chrome: Provides improved debugging support in Chrome with Elmish.
+1. **Configure VS Code** with the following extensions:
+    * [Ionide](https://marketplace.visualstudio.com/items?itemName=Ionide.Ionide-fsharp): Provides F# support to Code.
+    * [C#](https://marketplace.visualstudio.com/items?itemName=ms-vscode.csharp): Provides .NET Core debugging support.
+    * [Debugger for Chrome](https://marketplace.visualstudio.com/items?itemName=msjsdiag.debugger-for-chrome): Provides integrated client-side debugging in Code.
+
+## Launching and debugging a SAFE application in VS Code
+The SAFE template allows you to launch a SAFE Stack application with both client and server debugging enabled automatically.
+
+1. Create a new SAFE application using the [SAFE template](template-overview)
```
Comment (theimowski) [inline docs/feature-debugging.md]: make sure the relative link works, it doesn't for me locally with `mkdocs serve`

File: docs/feature-debugging.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+1. Open VS Code in the folder containing your application
+1. Go to the Debug pane (`CTRL + Shift + D`)
+ 
+    <center><img src="../img/feature-debugging-1.png" style="height: 400px;"/></center>
```
Comment (theimowski) [inline docs/feature-debugging.md]: probably not that important, but I think the default VS Code configuration comes with the Activity Bar at left hand side

File: docs/feature-debugging.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+1. Ensure that **Debug SAFE App** is selected as the launch configuration
+1. Hit `F5` to launch the application, or hit the green play button in the launch configuration panel
+
+That's it. VS Code will now automatically:
+
+* Download all dependencies for both client and server
+* Build both the client and server
+* Start both client and server
+* Attach debuggers for both the client and server simultaneously!
+
```
Comment (theimowski) [inline docs/feature-debugging.md]: Let's add a note that a browser will open, but one might need to refresh it after compilation ends

File: docs/feature-debugging.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+### Setting breakpoints
+To set a breakpoint, simply highlight the line of code that you wish to stop on and hit `F9`, or click on the left-side gutter of the code window; a red dot will appear to show the breakpoint is set. You can see more on VS Code breakpoints [here](https://code.visualstudio.com/docs/editor/debugging#_breakpoints).
+
+Here you can see a breakpoint that has been hit in the `update` method of the sample application that is bundled with the SAFE template.
+
+<center><img src="../img/feature-debugging-4.png" style="height: 200px;"/></center>
+
+### Viewing debug output
+Debug output from Chrome will automatically be sent to the integrated Debug Console. This is especially useful for Elmish applications, as you can interrogate message and model updates directly in the IDE.
```
Comment (theimowski) [inline docs/feature-debugging.md]: I'm not a native speaker and had to lookup `interrogate`, maybe just something simpler: `watch` or `trace`

File: docs/feature-debugging.md [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+<center><img src="../img/feature-debugging-6.png" style="height: 200px;"/></center>
+
+> Note: If you observe error messages regarding `SocketProtocolError` (see below), you can eliminate them by installing the Redux Dev Tools in the launched Chrome instance, as mentioned earlier. However, these errors can also safely be ignored.
```
Comment (theimowski) [inline docs/feature-debugging.md]: Let's write down the whole error message `Websocket connection to ws://localhost:8000/socketcluster/ failed: Error (...) response code: 404` - Search Engines might index that so people potentially find solution faster

File: docs/feature-debugging.md [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+## Restarting your application
+In the standard launch mode seen above, hot module reloading is enabled on the client, so changes that are made to your Fable application will occur immediately.
+
+However, in order to support debugging, watch mode is not enabled on the server; server changes require a restart of the server debugging session: Having ensured that Debug Server is selected in the floating Debug control, select the Restart icon or hit `CTRL + SHIFT + F5`. The server will stop, rebuild and restart with the debugger automatically re-attached.
+
+## Advanced options
+The SAFE template comes with several other modes of operation for debugging:
+
+### Full watch mode
+You can elect to use "watch" mode on both client *and* server by choosing the **Watch SAFE app** option from the Debug Launcher. Here, changes to any server file will automatically initiate a restart of it. However, whilst you will still be able to debug the client, in this mode there is no capability for server-side debugging.
```
Comment (theimowski) [inline docs/feature-debugging.md]: `elect` -> `select`

File: docs/feature-debugging.md [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+
+### Debug Server or Client only
+You can elect to launch and debug only the client or server in isolation. This may be useful if you are running the other component separately e.g. through FAKE, the `dotnet` command line tool directly or via Visual Studio. You can initiate this by choosing either **Debug Client** or **Debug Server** as required.
```
Comment (theimowski) [inline docs/feature-debugging.md]: as above

File: docs/feature-debugging.md [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+
+### Build Server
+You can initiate a restore and rebuild of the server by hitting `CTRL + SHIFT + B` and selecting **Build Server** from the drop-down.
```
Comment (theimowski) [inline docs/feature-debugging.md]: Maybe we should add a "Known issues" / "Troubleshooting" section at the very end listing:
1. issue with node process staying alive after debug session (with link to GH issue)
1. SocketProtocolError with ReduxDevTools
1. Chrome window starting before it's actually useful

**Comment by isaacabraham** [comment]
@theimowski I think that this is ready.

**Pull request merged**